pkg/ingester/ingester_v2.go

```diff
@@ -745,25 +741,17 @@ func (i *Ingester) v2Push(ctx context.Context, req *cortexpb.WriteRequest) (*cor
 
 			// If the cached reference exists, we try to use it.
 			if cachedRefExists {
-				if err = app.AddFast(cachedRef, s.TimestampMs, s.Value); err == nil {
+				if _, err = app.Append(cachedRef, copiedLabels, s.TimestampMs, s.Value); err == nil {
```

Comment:
Should we verify that returned `ref` is the same as `cachedRef`? If it isn't, it means that our `cachedRef` is obsolete, and should be updated in `refCache`. If we don't do that, we will keep reusing old ref, and not get any benefit from caching. I wonder if it is possible to test for this scenario.

Reply:
This is an excellent catch Peter! I added a test case for the same. Can you take a look?